toucan_connectors/common.py

```diff
@@ -103,6 +108,10 @@ def _handle_missing_params(elt, params):
                             Template('{{ %s }}' % m, undefined=StrictUndefined).render(params)
                         except Exception:
                             missing_params.append(m)
+                            if k == 'url':
+                                raise NonValidEndpointVariable(
+                                    f'Non valid variable {m} in endpoint'
+                                )
```

Comment:
This method `_handle_missing_params` is very generic. It's not directly linked to the HTTP connector in which the 'url' param is mandatory.
I would rather add an argument `mandatory_keys` to `_handle_missing_params`. The mandatory keys would not be allowed to fail the template rendering step, and would raise an error instead of ignoring it.
Then, the HTTP connector could use this method, saying that the `url` parameter is not allowed to fail.
What do you think?

Reply:
'url' is a little bit poorly named, it's actually a partial url like `/posts` rather than `https://example.com/posts`. It's not required.

It's not technically a mandatory key but I can pass an array of parameters with a different name perhaps :smile: